### ISSUE TITLE
[BUG]: aks-log-collector.sh creates a large ip_netns_commands.txt which lead to ephemeral-storage issues

### ISSUE THREAD

**Issue by akorp** [Azure/AKS#4148]
**Describe the bug**
We have observed that files `ip_netns_commands.txt` (example of location folder /tmp/tmp.4FKbTfOrn4/collect) in our AKS cluster nodes sometimes growing to many GBs and when the size comes to around 90GB nodes start having issues with ephemeral storage (`The node was low on resource: ephemeral-storage.`) then pods become evicted and multiple other issues appear.

```
root@aks-apps5-<redacted>-vmss000014:/tmp/tmp.4FKbTfOrn4/collect# ls -lh ip_netns_commands.txt 
-rw-r--r-- 1 root root 38G Mar  7 13:19 ip_netns_commands.txt
root@aks-apps5-<redacted>-vmss000014:/tmp/tmp.4FKbTfOrn4/collect# fuser -v ip_netns_commands.txt
                     USER        PID ACCESS COMMAND
/tmp/tmp.4FKbTfOrn4/collect/ip_netns_commands.txt:
                     root      83233 F.... ip
                     root      109814 F.... ip
                     root      560481 F.... ip
                     root      1133085 F.... ip
                     root      1133086 F.... ip
                     root      1133087 F.... ip
                     root      1134797 F.... ip
                     root      1737066 F.... ip
                     root      1737172 F.... ip
                     root      1737210 F.... ip
                     root      1737451 F.... ip
```

```
root@aks-apps5-<redacted>-vmss000014:/tmp/tmp.4FKbTfOrn4/collect# pstree -aps 83233
systemd,1
  └─aks-log-collect,61814 /opt/azure/containers/aks-log-collector.sh
      └─ip,83233 -all netns exec /bin/bash -x -c...
          └─ip,109814 -all netns exec /bin/bash -x -c...
              └─ip,1737066 -all netns exec /bin/bash -x -c...
                  └─ip,1737172 -all netns exec /bin/bash -x -c...
                      └─ip,1737210 -all netns exec /bin/bash -x -c...
                          └─ip,1737451 -all netns exec /bin/bash -x -c...
                              └─ip,560481 -all netns exec /bin/bash -x -c...
                                  └─ip,1133085 -all netns exec /bin/bash -x -c...
                                      └─bash,1147264 -x -c...
                                          └─ss,1147282 -anoempiO --cgroup
```

```
root@aks-apps5-<redacted>-vmss000014:/tmp/tmp.4FKbTfOrn4/collect# head --lines 20 /opt/azure/containers/aks-log-collector.sh
#! /bin/bash
#
# AKS Log Collector
#
# This script collects information and logs that are useful to AKS engineering
# for support and uploads them to the Azure host via a private API. These log
# bundles are available to engineering when customers open a support case and
# are especially useful for troubleshooting failures of networking or
# kubernetes daemons.
#
# This script runs via a systemd unit and slice that limits it to low CPU
# priority and 128MB RAM, to avoid impacting other system functions.

# Log bundle upload max size is limited to 100MB
MAX_SIZE=104857600

# Shell options - remove non-matching globs, don't care about case, and use
# extended pattern matching
shopt -s nullglob nocaseglob extglob
```

**Environment (please complete the following information):**
 - Kubernetes version [1.28.5]

**Comment by Noxumrjohnson** [comment]
Hi akorp, 
we have pretty the same problem - did you found a solution for that "BUG/Problem"?

**Comment by robdope** [comment]
Hi akorp,
did you found a solution for that problem/bug ?

my problem is that this file grows until the disk-space is reached and all pods etc. are switched to another node..
the growing file is: "/tmp/tmp.L7DmtYtrA8/collect/ip_netns_commands.txt"

best regards
rob

**Comment by akorp** [comment]
> Hi akorp, did you found a solution for that problem/bug ?
> 
> my problem is that this file grows until the disk-space is reached and all pods etc. are switched to another node.. the growing file is: "/tmp/tmp.L7DmtYtrA8/collect/ip_netns_commands.txt"
> 
> best regards rob

Hi @robdope

We opened a ticket with Microsoft and got an answer from them:
```
We were able to reproduce the issue on our end and have escalated it to the product team. We're currently waiting for their response, and I'll keep you informed as soon as we receive any updates.
```

We still have this issue in our test and production nodes when ip_netns_commands.txt file takes most of the node space.  As a workaround for now we empty the file with command like  `> ip_netns_commands.txt`.

**Comment by robdope** [comment]
> > Hi akorp, did you found a solution for that problem/bug ?
> > my problem is that this file grows until the disk-space is reached and all pods etc. are switched to another node.. the growing file is: "/tmp/tmp.L7DmtYtrA8/collect/ip_netns_commands.txt"
> > best regards rob
> 
> Hi @robdope
> 
> We opened a ticket with Microsoft and got an answer from them:
> 
> ```
> We were able to reproduce the issue on our end and have escalated it to the product team. We're currently waiting for their response, and I'll keep you informed as soon as we receive any updates.
> ```
> 
> We still have this issue in our test and production nodes when ip_netns_commands.txt file takes most of the node space. As a workaround for now we empty the file with command like `> ip_netns_commands.txt`.

Hi @akorp,
ok good - thank you for the infos, good to know about the ticket with MS and your workaround.
Out workaround is at the moment to recreate the nodes (cordon -> drain) - after that, the ip_netns_commands.txt doesnt extists on the node and the discspace isnt growing anymore... for today it works, we'll see tomorrow..
thank you so far

**Comment by robdope** [comment]
Hi @akorp ,
we resolved the problem with "Reimage" the VMs in the VMSS. 
In this post its explained: https://stackoverflow.com/questions/70668793/azure-vmss-instance-latest-model-meaning
maybe it help you too.

**Comment by akorp** [comment]
@robdope 
Yes, we got an answer from Microsoft today that the latest image `202403.04.0` should solve this issue.

**Comment by akorp** [comment]
Seems like the issue is still there. We have our node pool on version `AKSUbuntu-2204gen2containerd-202403.04.0`, and still the file `ip_netns_commands.txt` is 29GB.

**Comment by akorp** [comment]
The issue is still there on the our latest available node image `AKSUbuntu-2204gen2containerd-202404.09.0`. The case with Microsoft is also still open.

**Comment by UtheMan** [comment]
One way to stop the issue is disabling log collector on the nodes, it is controlled by a timer systemd unit. 

Following commands should do the job:
```bash
systemctl stop aks-log-collector.timer
systemctl disable aks-log-collector.timer
```
This would have to be ran on every node, so a daemonset running that might be handy.
We are looking into a "long term" fix for this issue.

**Comment by akorp** [comment]
The workaround doesn't seems work. Probably because there is an old process still running.

```
❯ date
Mon Apr 29 09:29:12 CEST 2024
❯ kubectl aks run-command "du -h /tmp/tmp.e8ZVqZ7yzb/collect/ip_netns_commands.txt" --node aks-apps3-24490056-vmss00001g
42G	/tmp/tmp.e8ZVqZ7yzb/collect/ip_netns_commands.txt
❯ kubectl aks run-command "> /tmp/tmp.e8ZVqZ7yzb/collect/ip_netns_commands.txt" --node aks-apps3-24490056-vmss00001g
❯ kubectl aks run-command "systemctl stop aks-log-collector.timer" --node aks-apps3-24490056-vmss00001g
❯ kubectl aks run-command "systemctl disable aks-log-collector.timer" --node aks-apps3-24490056-vmss00001g
Removed /etc/systemd/system/timers.target.wants/aks-log-collector.timer.
```

```
root@aks-apps3-24490056-vmss00001G:~# date
Thu May  2 10:31:22 UTC 2024
root@aks-apps3-24490056-vmss00001G:~# systemctl status aks-log-collector.timer
○ aks-log-collector.timer - AKS Log Collector Timer
     Loaded: loaded (/etc/systemd/system/aks-log-collector.timer; disabled; vendor preset: enabled)
     Active: inactive (dead)
    Trigger: n/a
   Triggers: ● aks-log-collector.service

Apr 23 02:39:33 aks-apps3-24490056-vmss00001G systemd[1]: Started AKS Log Collector Timer.
Apr 29 07:37:39 aks-apps3-24490056-vmss00001G systemd[1]: aks-log-collector.timer: Deactivated successfully.
Apr 29 07:37:39 aks-apps3-24490056-vmss00001G systemd[1]: Stopped AKS Log Collector Timer.
root@aks-apps3-24490056-vmss00001G:~# du -h /tmp/tmp.e8ZVqZ7yzb/collect/ip_netns_commands.txt
641M/tmp/tmp.e8ZVqZ7yzb/collect/ip_netns_commands.txt
root@aks-apps3-24490056-vmss00001G:~# fuser -v /tmp/tmp.e8ZVqZ7yzb/collect/ip_netns_commands.txt
                     USER        PID ACCESS COMMAND
/tmp/tmp.e8ZVqZ7yzb/collect/ip_netns_commands.txt:
                     root      67471 F.... ip
                     root      67472 F.... ip
                     root      68955 F.... ip
                     root      68956 F.... ip
                     root      68957 F.... ip
                     root      68958 F.... ip
                     root      68959 F.... ip
....
root@aks-apps3-24490056-vmss00001G:~# pstree -aps 67471
systemd,1
  └─aks-log-collect,51884 /opt/azure/containers/aks-log-collector.sh
      └─ip,67471 -all netns exec /bin/bash -x -c...
          └─ip,67472 -all netns exec /bin/bash -x -c...
              └─ip,68955 -all netns exec /bin/bash -x -c...
                  └─ip,68956 -all netns exec /bin/bash -x -c...
                      └─ip,68957 -all netns exec /bin/bash -x -c...
                          └─ip,68958 -all netns exec /bin/bash -x -c...
                              └─ip,68959 -all netns exec /bin/bash -x -c...
...
```

**Comment by UtheMan** [comment]
@akorp Is it possible for you to stop these old processes? Like you mentioned old process that is still running is most likely the reason, because the timer systemd unit is what controls that log collector.

We have a permanent fix ready and merged, it will start releasing later next week. I will update this thread with the node image version that will have the fix. Once it reaches you region you will then need to perform a node image upgrade on your affected nodes.

**Comment by akorp** [comment]
@UtheMan thanks for the update 😃 We have disabled the process and looking forward to the new image.

**Comment by UtheMan** [comment]
@akorp the image with fix for log collector started releasing to prod regions and should be available globally within 2~ weeks. You can track the release [here](https://releases.aks.azure.com/) (AKS Node Images tab on the left side). 
The node image containing the fix is `202405.03.0`. Once it reaches your region perform a node image upgrade.
